pkg/interface/chat/src/js/api.js

```diff
@@ -28,131 +73,245 @@ class UrbitApi {
-        }, 
-        (err) => {
-          reject(err);
-        });
+  groupRemove(members, path) {
+    this.groupsAction({
+      remove: {
+        members, path
+      }
     });
   }
 
-  notify(aud, bool) {
-    this.hall({
-      notify: {
-        aud,
-        pes: !!bool ? 'hear' : 'gone'
+  chatAction(data) {
+    this.action("chat-store", "json", data);
+  }
+
+  chatCreate(path, owner = `~${window.ship}`) {
+    this.chatAction({
+      create: {
+        path, owner
       }
     });
   }
 
-  permit(cir, aud, message) {
-    this.hall({
-      permit: {
-        nom: cir,
-        sis: aud,
-        inv: true
+  chatDelete(path) {
+    this.chatAction({ delete: { path } });
+  }
+
+  chatMessage(path, author, when, letter) {
+    let data = {
+      message: {
+        path,
+        envelope: {
+          uid: uuid(),
+          number: 0,
+          author,
+          when,
+          letter
+        }
+      }
+    };
+
+    this.chatHookAction(data, "json");
+    this.addPendingMessage(data.message);
+  }
+
+  chatRead(path, read) {
+    this.chatAction({ read: { path } });
+  }
+
+  chatViewAction(data) {
+    console.log(data);
+    this.action("chat-view", "json", data);
+  }
+
+  chatViewCreate(path, security, read, write) {
+    this.chatViewAction({
+      create: {
+        path, security, read, write
       }
     });
+  }
 
-    if (message) {
-      this.invite(cir, aud);
-    }
+  chatViewDelete(path) {
+    this.chatViewAction({ delete: { path } });
   }
 
-  unpermit(cir, ship) {
-    /*
-     * lol, never send an unpermit to yourself.
-     * it puts your ship into an infinite loop.
-     * */
-    if (ship === window.ship) {
-      return;
-    }
-    this.hall({
-      permit: {
-        nom: cir,
-        sis: [ship],
-        inv: false
+  inviteAction(data) {
+    console.log(data);
+    this.action("invite-store", "json", data);
+  }
+
+  inviteCreate(path) {
+    this.inviteAction({ create: { path } });
+  }
+
+  inviteDelete(path) {
+    this.inviteAction({ delete: { path } });
+  }
+
+  inviteAccept(path, uid) {
+    this.inviteAction({ accept: { path, uid } });
+  }
+
+  inviteDecline(path, uid) {
+    this.inviteAction({ decline: { path, uid } });
+  }
+
+  inviteInvite(path, peerPath, ship, app, text, recipient) {
+    this.inviteAction({
+      invite: {
+        path, 
+        invite: {
+          'peer-path': peerPath,
+          dock: { ship, app },
+          text, recipient
+        },
+        uid: uuid()
       }
     });
   }
 
-  invite(cir, aud) {
-    let audInboxes = aud.map((aud) => `~${aud}/i`);
-    let inviteMessage = {
-      aud: audInboxes,
-      ses: [{
-        inv: {
-          inv: true,
-          cir: `~${window.ship}/${cir}`
-        }
-      }]
+  chatHookAction(data, mark = "chat-hook-action") {
+    this.action("chat-hook", mark, data);
+  }
+
+  chatHookAddOwned(path, security) {
+    let data = {};
+    data['add-owned'] = {
+      path, security
+    };
```

Comment:
This is fine, but FWIW it might be simpler to just do:

```
let data = {
  add-owned: { path, security }
}
```

here and below.

Reply:
Maybe my syntax highlighter is weird, but I always thought that didn’t work.

Reply:
You might be right in that you may need quotes around `add-owned`, but otherwise it's ok.

Reply:
aesthetic, closing